app/src/main/java/org/sinisterstuf/guesstheanimal/ui/GuessActivity.java

```diff
@@ -56,50 +69,34 @@ public void parseYes() {
 		}
 	}
 
-	/**
-	 * Called when the user chooses No
-	 */
 	@OnClick(R.id.noButton)
 	public void parseNo() {
 		if (finalGuess) {
-			learnNewAnimal(prevReq);
+			learnNewAnimal();
 		} else {
 			parseChoice(false, animal.noAnimal);
 		}
 	}
 
-	/**
-	 * Compares the user's <code>choice</code> to the desired answer for this
-	 * <code>Animal</code> from <code>Animal.answerWhenMe</code>
-	 *
-	 * @param choice the user's choice
-	 */
 	private void parseChoice(boolean choice, Animal next) {
 		if (choice == animal.answerWhenMe) {
-			Intent intent = new Intent(this, GuessActivity.class);
-			intent.putExtra(Animal.ANIMAL, animal);
-			intent.putExtra(FINAL_GUESS, true);
-			startActivity(intent);
+			GuessActivity.start(this, animal, true, false);
 		} else {
 			if (next == null) {
-				learnNewAnimal(choice);
+				learnNewAnimal();
 			} else {
-				Intent intent = new Intent(this, GuessActivity.class);
-				intent.putExtra(Animal.ANIMAL, next);
-				intent.putExtra(Animal.NEXT_REQ, choice);
-				intent.putExtra(FINAL_GUESS, false);
-				startActivity(intent);
+				GuessActivity.start(this, next, false, choice);
 			}
 		}
 	}
 
 	/**
 	 * Start a new activity to learn an animal
 	 */
-	private void learnNewAnimal(Boolean nextReq) {
+	private void learnNewAnimal() {
```

Comment:
This part is not OK. The learnNewAnimal method is only _sometimes_ called with `prevReq` (the user's response to the _previous_ question) as an argument, the rest of the time it's called with `choice` which is the result of their _current_ interaction. So it's not as simple as just use the value we have in prevReq 😉 I'll leave this part out of the PR when applying this patch.